apps/prairielearn/elements/pl-rich-text-editor/pl-rich-text-editor.js

```diff
@@ -94,4 +94,38 @@
   MathFormula.tagName = 'SPAN';
 
   Quill.register('formats/formula', MathFormula, true);
+
+  class Counter {
+    constructor(quill, options) {
+      this.quill = quill;
+      this.options = options;
+      this.container = document.querySelector(options.container);
+      if (this.options.unit !== 'none') {
+        quill.on(Quill.events.TEXT_CHANGE, this.update.bind(this));
+      }
+    }
+
+    calculate() {
+      const text = this.quill.getText();
+
+      // FIXME: Support unicode characters in character count
+      if (this.options.unit === 'word') {
+        const trimmed = text.trim();
+        // Splitting empty text returns a non-empty array
+        return trimmed.length > 0 ? trimmed.split(/\s+/).length : 0;
+      } else if (this.options.unit === 'character') {
+        return text.length;
```

Comment:
To handle Unicode characters:
```suggestion
        return [...text].length;
```

Reply:
I am not sure if spread count is 100% accurate

```js
[..."😍🦶🏾"].length ==> 3
```

Reply:
It's still better than byte count, so I think I'll use this.

Reply:
> I am not sure if spread count is 100% accurate
> 
> ```js
> [..."😍🦶🏾"].length ==> 3
> ```

That's still technically 3 characters. If you hit backspace on the foot you get a different icon, since that foot is a combination of two characters.

Reply:
At least on firefox, backspace deletes the foot. CLion on the other hand deletes the color and leaves the foot behind